.appveyor.yml

```diff
@@ -0,0 +1,110 @@
+  NP_BUILD_DEP: "numpy==1.10.4"
+  NP_TEST_DEP: "numpy==1.10.4"
+
+  # SDK v7.0 MSVC Express 2008's SetEnv.cmd script will fail if the
+  # /E:ON and /V:ON options are not enabled in the batch script intepreter
+  # See: http://stackoverflow.com/a/13751649/163740
+  CMD_IN_ENV: "cmd /E:ON /V:ON /C .\\appveyor\\run_with_env.cmd"
+
+  matrix:
+    - PYTHON: C:\Python27-x64
+      PYTHON_ARCH: "64"  # define this to download 64-bit FFTW dlls
+      PYTHON_VERSION: "2.7"  # used by run_with_env.cmd
+    - PYTHON: C:\Python35-x64
+      PYTHON_ARCH: "64"
+      PYTHON_VERSION: "3.5"
+    - PYTHON: C:\Python36-x64
+      PYTHON_ARCH: "64"
+      PYTHON_VERSION: "3.6"
+      NP_BUILD_DEP: "numpy==1.12"
+      NP_TEST_DEP: "numpy==1.12"
+    - PYTHON: C:\Python37-x64
+      PYTHON_ARCH: "64"
+      PYTHON_VERSION: "3.7"
+      NP_BUILD_DEP: "numpy==1.14.5"
+      NP_TEST_DEP: "numpy==1.14.5"
+
+init:
+  - "ECHO \"%APPVEYOR_SCHEDULED_BUILD%\""
+
+install:
+
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside the powershell script as it would require to restart
+  # the parent CMD process).
+  - SET PATH=%PYTHON%;%PYTHON%\Scripts;%PATH%
+
+  # Check that we have the expected version and architecture for Python
+  - python --version
+  - python -c "import struct; print(struct.calcsize('P') * 8)"
+
+  - python -m pip install -U pip
+
+  # Pin wheel to 0.26 to avoid Windows ABI tag for built wheel
+  - pip install wheel==0.26
+  - dir %REPO_DIR%  # REPO_DIR should be empty prior to submodule update
+  - git submodule update --init
+
+  - cd %REPO_DIR%
+  # Get and configure the FFTW libs (must occur after the submodule update)
+  - "%CMD_IN_ENV% ./appveyor/setup_fftw_dlls.cmd"
+
+  # - cd %REPO_DIR%
+  # - git checkout %BUILD_COMMIT%
+
+  # Install the build dependencies of the project.
+  - pip install %NP_TEST_DEP%
+  - pip install -r requirements.txt
+
+build_script:
+  # Build the wheel
+  - git fetch origin
+  - ECHO %BUILD_COMMIT%
+  - git checkout %BUILD_COMMIT%
+  - git status  # use to verify we are at the expected tag
+  # - set PYFFTW_LIB_DIR=C:\projects\pyfftw-wheels\%REPO_DIR%
+  - python -c "from pkg_resources import get_build_platform; print(get_build_platform())"  # should be win-amd64
+  # - python setup.py bdist_wheel
+  - "%CMD_IN_ENV% python setup.py -v bdist_wheel"
+  - "%CMD_IN_ENV% python setup.py -v build_ext --inplace"
+  - ps: |
+      # Upload artifact to Appveyor immediately after build
+      ls dist -r | Foreach-Object {
+          appveyor PushArtifact $_.FullName
+          pip install $_.FullName
+      }
+
+test_script:
+  # Install the test / runtime dependencies
+  - pip install %NP_TEST_DEP%
+  # Install the generated wheel package to test it
+  - pip install --pre --no-index --find-links .\%REPO_DIR%\dist %PKG_NAME%
+  # Check Python version just in case
+  - python --version
+  # Run unit tests with pytest
+  - "python setup.py test"
```

Comment:
Should we cleanup the downloaded FFTW DLLs before this step to ensure it is picking the ones in the wheel up?

Reply:
Usually you can run the tests as long as the package is installed somewhere the Python interpreter can find it. So would try `pip install`ing the wheel, cleaning the source directory, and running the tests with `python -m unittest discover`.

Reply:
That said, it's not obvious to me if we are running the tests for the other platforms. So maybe this is good enough as is.

Reply:
Tests are not currently being run for OS X or linux. That is issue #12 and was being worked on in #11.

Reply:
I think we have more variants in this repo than we test in the main pyFFTW one, so it would probably be worth having the tests all running here as well at some point.

Reply:
I tried testing as you suggest locally (i.e. installing from the wheel, cleaning the source directory with `git clean -xfd` and then running tests via `python -m unittest discover`. It finds the tests, but I get a ton of `ImportErrors`.

```
C:\Users\Greg\src\pyFFTW>python -m unittest discover
EEEEEEEEEEEEEEEEEE
======================================================================
ERROR: pyfftw (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: pyfftw
Traceback (most recent call last):
  File "F:\Miniconda3\lib\unittest\loader.py", line 462, in _find_test_path
    package = self._get_module_from_name(name)
  File "F:\Miniconda3\lib\unittest\loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "C:\Users\Greg\src\pyFFTW\pyfftw\__init__.py", line 18, in <module>
    from .pyfftw import (
ImportError: No module named 'pyfftw.pyfftw'


======================================================================
ERROR: test.test_pyfftw_base (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test.test_pyfftw_base
Traceback (most recent call last):
  File "F:\Miniconda3\lib\unittest\loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "F:\Miniconda3\lib\unittest\loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "C:\Users\Greg\src\pyFFTW\test\test_pyfftw_base.py", line 35, in <module>
    from pyfftw import FFTW, _supported_types, _all_types_human_readable
  File "C:\Users\Greg\src\pyFFTW\pyfftw\__init__.py", line 18, in <module>
    from .pyfftw import (
ImportError: No module named 'pyfftw.pyfftw'
```

If I first (`cd test` to move out of the root source directory and into the `test` subfolder) I get similar errors, but this time about invalid relative imports.
```
======================================================================
ERROR: test_pyfftw_builders (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_pyfftw_builders
Traceback (most recent call last):
  File "F:\Miniconda3\lib\unittest\loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "F:\Miniconda3\lib\unittest\loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "C:\Users\Greg\src\pyFFTW\test\test_pyfftw_builders.py", line 38, in <module>
    from .test_pyfftw_base import run_test_suites, require
ImportError: attempted relative import with no known parent package
```